src/App.jsx

```diff
@@ -1,20 +1,36 @@
 import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
+import React from 'react'
 import './App.css'
 import TodoList from './TodoList'
 import AddTodoForm from './AddTodoForm'
 
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem('savedTodoList')));
```

Comment:
The way you are setting the state would lead to bug eventually when there 'localStorage.getItem('savedTodoList')' return null.

Remember, you are passing the 'todoList' state variable to the <TodoList  /> component [here](https://github.com/pascascio/react-todo/blob/16eb8a3588042dfd686a1667a14ffab5b6d6aa52/src/App.jsx#L31) which assume the 'todoList' variable is an array and when it's accessing 'map' property on the array [here](https://github.com/pascascio/react-todo/blob/16eb8a3588042dfd686a1667a14ffab5b6d6aa52/src/TodoList.jsx#L9) it would return an error because 'map' property does not exist on null